posthog/auth.py

```diff
@@ -234,6 +234,15 @@ def authenticate(self, request: Union[HttpRequest, Request]) -> Optional[tuple[A
             if team is None:
                 return None
 
+            # NB: This import needs to be here and not at module level to avoid a circular import
+            from posthog.api.utils import get_token
+
+            # Security check: Ensure any project API key in request body matches this team
+            request_data = request.data if hasattr(request, "data") else None
+            project_api_token = get_token(request_data, request)
+            if project_api_token and project_api_token != team.api_token:
+                raise AuthenticationFailed("Project API key does not match authenticated project")
+
```

Comment:
This is causing a lot of test failures. While I think this is a valid check, we may want to limit it to only `local_evaluation` and `remote_config`

Reply:
Actually, this is duplicating logic elsewhere and is incorrect. I'll remove it.